### DISCUSSION TITLE
Option for Commands Without Directly Inserting Them Into the Input

### DISCUSSION THREAD

**Discussion by davidsuarezcdo** [open-webui/open-webui#6029] [Ideas]
Currently, in Open WebUI, when typing the / symbol and selecting a command, it is directly inserted into the input field. This can be useful in scenarios where the user wants to modify or customize the command. However, in some cases, it would be more convenient if the command is not added to the input but rather remains as a visual reference above the input field, similar to how platforms like Taskade handle it.

**Proposed Functionality:**

- When typing / in the input field, a dropdown menu appears with available commands.
- The user selects the desired command.
- Instead of inserting the command directly into the input, it is shown as an "attachment" or "surface" above the input field, where the selected command and any relevant parameters can be viewed.
- The user can execute the command without modifying it, or if desired, they can choose to expand and modify it before execution.

**Benefits:**

- Keeps the input field clean and clear, allowing users to have full control over what they actually type.
- Facilitates quick command execution without the need to edit the input field.
- Provides a more intuitive and efficient experience, especially in cases where modifying the predefined command is unnecessary.

**Technical Proposal:** This could be implemented by adding an option where selected commands behave as "hints" or additional "surfaces" anchored right above the input field, without interfering with what the user types.

Example:

![image](https://github.com/user-attachments/assets/77434693-a479-4bad-ad79-2d10ef474601)

**Comment by frettig-radiox** [comment]
I completely support this proposal. It makes perfect sense to have selected commands behave more like "hints" or "surfaces" above the input field. There's an ongoing discussion about this topic over at [Open WebUI Discussions #5024](https://github.com/open-webui/open-webui/discussions/5024). Feel free to share your thoughts there as well! Frank
